tools/nb_code_sync.py

```diff
@@ -123,6 +123,8 @@ def _is_source_code_equal(x_list, y_list):
 
   def _set_cell_source(self, cell_id, source):
     for i, cell in enumerate(self.data["cells"]):
+      if "id" not in cell["metadata"]:
```

Comment:
I know this is not a translation, but I wanted to leave a quick suggestion. How about putting everything in a `try` statement?

```
try:
    if cell["metadata"]["id"] == cell_id:
        self.data["cells"][i]["source"] = source
        break
except RuntimeError: # Or any other error that the code block produces
    cell["metadata"]["id"] = ""
```